### PR TITLE
Configure jest for testing

### PR DESCRIPTION
**Pull request by johnwayodi**
#### Description
- move some modules to devDependencies
- improve on env configuration
- add jest scripts and configs
- add linter and commands
- add morgan for logging requests
- add nodemon for dev purposes
#### Type of change

Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Others (cosmetics, styling, improvements)
- [x] This change requires a documentation update

#### How Has This Been Tested?

- [ ] Unit
- [ ] Integration
- [ ] End-to-end

#### How to Test
1. checkout to the branch, delete existing `node_modules`, and do a clean install of dependencies `yarn install`
2. run tests using the command `yarn test`
3. run specific test file using the command `yarn test test-file-path`
4. check linting errors `yarn lint`
5. fix linting errors `yarn lint:fix`

#### Any background context you want to add?

 [Resolves Issue #3]

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -0,0 +1,5 @@
+NODE_ENV=
+APP_PORT=
+REDIS_URL=
+SLACK_WEBHOOK_URL=
+SLACK_SIGNING_SECRET=
```
Comment (Allan690) [inline .env.sample]: Perhaps you could add new line at the end of this and the nodemon.json file

File: nodemon.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "watch": ["src"],
+  "ext": "ts",
+  "ignore": ["src/**/*.spec.ts"],
+  "exec": "ts-node-dev -r tsconfig-paths/register src/index.ts"
```
Comment (Allan690) [inline nodemon.json]: Would have loved to see the inclusion of the `--transpileOnly` flag which according to documentation of ts-node-dev [here](https://www.npmjs.com/package/ts-node-dev) speeds up things greatly.  You can also include the `--inspect --` flag to enable chrome debugging in dev mode

File: package.json [hunk 3]
```diff
@@ -4,32 +4,73 @@
...
+    "clean": "rimraf dist",
+    "tsc": "ts-node-dev -r tsconfig-paths/register",
+    "prebuild": "yarn clean",
+    "build": "tsc -b tsconfig.build.json",
+    "jest": "ts-node-dev ./node_modules/jest/bin/jest.js",
+    "test": "yarn jest --coverage",
+    "start:dev": "NODE_ENV=development DEBUG=log nodemon",
+    "start:prod": "yarn build && DEBUG=log node dist/index.js",
+    "lint": "tslint -c tslint.json -p tsconfig.json",
+    "lint:fix":"yarn lint --fix"
```
Comment (Allan690) [inline package.json:16]: Nice work on separating these commands 👍

File: package.json [hunk 4]
```diff
@@ -4,32 +4,73 @@
...
+    "ts-jest": "^24.1.0",
     "ts-node-dev": "^1.0.0-pre.43",
+    "tsconfig-paths": "^3.9.0",
     "tslint": "^5.20.0",
-    "tslint-config-prettier": "^1.18.0",
-    "tsscmp": "^1.0.6",
+    "tslint-config-airbnb": "^5.11.2",
     "typescript": "^3.6.3"
+  },
+  "jest": {
```
Comment (Allan690) [inline package.json:52]: 💪

File: src/cache/index.ts [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import LRUCacheSingleton from './lruCache';
 import RedisCacheSingleton from './redisCache';
+import env from '../config/environment';
```
Comment (Allan690) [inline src/cache/index.ts:3]: Nice stuff

**Review by Allan690 — COMMENTED** [review]
Nice work man, a few comments

File: nodemon.json (hunk 2 above)
Comment (johnwayodi) [reply nodemon.json]: @Allan690 , I have made the changes and included the flags.

File: .env.sample (hunk 1 above)
Comment (johnwayodi) [reply .env.sample]: Done

**Review by Allan690 — APPROVED** [review]

**Pull request merged**